test/utils/deployment.go

```diff
@@ -78,36 +78,34 @@ func waitForDeploymentCompleteMaybeCheckRolling(c clientset.Interface, d *apps.D
 		reason     string
 	)
 
-	err := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
-		var err error
-		deployment, err = c.AppsV1().Deployments(d.Namespace).Get(context.TODO(), d.Name, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-
-		// If during a rolling update, make sure rolling update strategy isn't broken at any times.
-		if rolling {
-			reason, err = checkRollingUpdateStatus(c, deployment, logf)
+	err := wait.PollUntilContextTimeout(context.Background(), pollInterval, pollTimeout, true,
```

Comment:
can we keep the function signature in just one (admittedly long) line? that would make the review much easier because only that line should change

Reply:
Other places too